File: convert_excel_to_json.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import os
import sys
import json
import shutil
import pandas as pd
from datetime import datetime

# 配置信息
excel_file_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data', '修改器合集.xlsx')
json_output_path = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data', 'modifiers_data.json')
backup_dir = os.path.join(os.path.dirname(os.path.abspath(__file__)), 'data', 'backups')

print('===== Excel 到 JSON 转换工具 (Python版) =====')
print('此工具将自动将 \'修改器合集.xlsx\' 转换为 \'modifiers_data.json\'')
print('==================================')
print(f'当前工作目录: {os.path.dirname(os.path.abspath(__file__))}')

# 检查Excel文件是否存在
if not os.path.exists(excel_file_path):
    print('错误: Excel文件不存在!')
    print(f'预期路径: {excel_file_path}')
    input('\n按回车键退出...')
    sys.exit(1)

# 检查pandas库是否已安装
try:
    import pandas as pd
except ImportError:
    print('\n错误: 未找到pandas库!')
    print('请先安装pandas库，然后再运行此脚本。')
    print('安装方法: 在终端中运行 "pip3 install pandas openpyxl"')
    input('\n按回车键退出...')
    sys.exit(1)

# 创建备份目录
if not os.path.exists(backup_dir):
    os.makedirs(backup_dir)
    print(f'已创建备份目录: {backup_dir}')

# 创建当前JSON文件的备份（如果存在）
if os.path.exists(json_output_path):
    backup_file_name = f'modifiers_data_backup_{datetime.now().strftime("%Y-%m-%d_%H-%M-%S")}.json'
    backup_path = os.path.join(backup_dir, backup_file_name)
    shutil.copy2(json_output_path, backup_path)
    print(f'已创建备份: {backup_path}')

try:
    print('\n开始转换...')
    print(f'读取Excel文件: {excel_file_path}')
    
    # 读取Excel文件
    df = pd.read_excel(excel_file_path)
    
    # 转换为JSON
    json_data = df.to_json(orient='records', force_ascii=False, indent=2)
    
    # 写入JSON文件
    with open(json_output_path, 'w', encoding='utf-8') as f:
        f.write(json_data)
    
    print(f'转换成功!')
    print(f'JSON文件已保存到: {json_output_path}')
    print(f'共转换 {len(df)} 条记录')
    print('\n✅ 转换完成!')
    
except Exception as e:
    print(f'\n❌ 转换过程中出错: {str(e)}')
    print('请检查Excel文件格式是否正确。')

print('\n按回车键退出...')
input()

File: convert_excel_to_json.js
const fs = require('fs');
const path = require('path');
const XLSX = require('xlsx');

// 配置信息
const EXCEL_FILE_PATH = path.join(__dirname, 'data', '修改器合集.xlsx');
const JSON_OUTPUT_PATH = path.join(__dirname, 'data', 'modifiers_data.json');
const BACKUP_DIR = path.join(__dirname, 'data', 'backups');

console.log('开始转换Excel到JSON...');
console.log(`读取Excel文件: ${EXCEL_FILE_PATH}`);

// 检查Excel文件是否存在
if (!fs.existsSync(EXCEL_FILE_PATH)) {
    console.error('错误: Excel文件不存在!');
    process.exit(1);
}

// 创建备份目录
if (!fs.existsSync(BACKUP_DIR)) {
    fs.mkdirSync(BACKUP_DIR);
}

// 创建当前JSON文件的备份（如果存在）
if (fs.existsSync(JSON_OUTPUT_PATH)) {
    const backupFileName = `modifiers_data_backup_${new Date().toISOString().replace(/[:.]/g, '-')}.json`;
    const backupPath = path.join(BACKUP_DIR, backupFileName);
    fs.copyFileSync(JSON_OUTPUT_PATH, backupPath);
    console.log(`已创建备份: ${backupPath}`);
}

try {
    // 读取Excel文件
    const workbook = XLSX.readFile(EXCEL_FILE_PATH);
    
    // 获取第一个工作表
    const firstSheetName = workbook.SheetNames[0];
    const worksheet = workbook.Sheets[firstSheetName];
    
    // 转换为JSON
    const jsonData = XLSX.utils.sheet_to_json(worksheet, { header: 1 });
    
    // 提取表头
    const headers = jsonData[0];
    
    // 转换数据格式，使每条记录都是对象
    const formattedData = jsonData.slice(1).map(row => {
        const record = {};
        headers.forEach((header, index) => {
            record[header] = row[index];
        });
        return record;
    });
    
    // 写入JSON文件
    fs.writeFileSync(JSON_OUTPUT_PATH, JSON.stringify(formattedData, null, 2));
    
    console.log(`转换成功!`);
    console.log(`JSON文件已保存到: ${JSON_OUTPUT_PATH}`);
    console.log(`共转换 ${formattedData.length} 条记录`);
    console.log('\n请按回车键退出...');
    
    // 等待用户按回车键
    process.stdin.resume();
    process.stdin.on('data', process.exit.bind(process, 0));
    
} catch (error) {
    console.error('转换过程中出错:', error.message);
    console.log('\n请按回车键退出...');
    process.stdin.resume();
    process.stdin.on('data', process.exit.bind(process, 1));
}